pkgs/development/compilers/miranda/default.nix

```diff
@@ -0,0 +1,41 @@
+{ stdenv, fetchurl, gcc }:
+
+stdenv.mkDerivation rec {
+  version = "2.066";
+  pname = "miranda";
+
+  src = fetchurl {
+    url = "https://www.cs.kent.ac.uk/people/staff/dat/miranda/src/mira-${builtins.replaceStrings [ "." ] [ "" ] version}-src.tgz";
+    sha256 = "1sqn031agwwgq3pka55pxd6m22dagmkxjhbwrni8gpigihg2h72j";
+  };
+
+  dontStrip = true;
+
+  enableParallelBuilding = true;
+
+  configurePhase = ''
+    patchShebangs quotehostinfo
+  '';
+
+  makeFlags = [ "CFLAGS=-O2" ];
+
+  buildFlags = [ "cleanup" "all" ];
+
+  installPhase = ''
+    mkdir -p $out/bin
+    mkdir -p $out/lib
+    mkdir -p $out/share/man/man1
+    install mira $out/bin
+    install mira.1 $out/share/man/man1
+    cp -r miralib $out/lib
+  '';
+
+  buildInputs = [ gcc ];
+
+  meta = with stdenv.lib; {
+    description = "Compiler for Miranda -- a pure, non-strict, polymorphic, higher order functional programming language";
+    homepage = "https://www.cs.kent.ac.uk/people/staff/dat/miranda/";
+    license = licenses.unfree;
```

Comment:
How did you determine this? The miralib/COPYING lppks like https://opensource.org/licenses/BSD-2-Clause.